### PR TITLE
Add dead modules code to MkFitEventOfHitsProducer

### PR DESCRIPTION
**Pull request by cerati**
This PR supersedes #24 , re-working the implementation in `MkFitEventOfHitsProducer`. See the original PR for more details.

### CODE REVIEW

File: RecoTracker/MkFit/plugins/MkFitEventOfHitsProducer.cc [hunk 1]
```diff
@@ -63,6 +73,27 @@ void MkFitEventOfHitsProducer::produce(edm::StreamID iID, edm::Event& iEvent, co
...
+  const auto& badStrips = siStripQuality.getBadComponentList();
+  for (const auto& bs : badStrips) {
+    const auto& surf = trackerGeom.idToDet(DetId(bs.detid))->surface();
+    const DetId detid(bs.detid);
+    const auto subdet = detid.subdetId();
+    const auto layer = mkFitGeom.topology()->layer(detid);
+    const auto isStereo = mkFitGeom.topology()->isStereo(detid);
+    bool isBarrel = (mkFitGeom.topology()->side(detid) == static_cast<unsigned>(TrackerDetSide::Barrel));
+    bool isPlusSide = (mkFitGeom.topology()->side(detid) == static_cast<unsigned>(TrackerDetSide::PosEndcap));
+    const auto ilay = mkFitGeom.layerNumberConverter().convertLayerNumber(subdet, layer, false, isStereo, isPlusSide);
```
Comment (makortel) [inline RecoTracker/MkFit/plugins/MkFitEventOfHitsProducer.cc:88]: This (and many of the intermediate variables) could be replaced with
```suggestion
    const auto ilay = mkFitGeom.mkFitLayerNumber(detid);
```

Comment (cerati) [reply RecoTracker/MkFit/plugins/MkFitEventOfHitsProducer.cc:88]: thanks, I am updating the PR

**Comment by makortel** [comment]
Thanks, the rest looks ok to me (given the list of improvement ideas in #24). I'll merge after https://github.com/trackreco/mkFit/pull/319 is merged (assuming no further conflicts).

**Comment by makortel** [comment]
Following the merge of https://github.com/trackreco/mkFit/pull/319

**Pull request merged**